pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/SinksImpl.java

```diff
@@ -744,6 +745,9 @@ public SinkConfig getSinkInfo(final String tenant,
 
         SinkConfigUtils.ExtractedSinkDetails sinkDetails = SinkConfigUtils.validateAndExtractDetails(
                 sinkConfig, classLoader, worker().getWorkerConfig().getValidateConnectorConfig());
+        if (classLoader instanceof URLClassLoader) {
```

Comment:
What about creating a utility method in some 'common' package.

I would also use Closeable and not URLClassloader